Error calculating the calendar splits of a job when RUNNING: ONCE

<!--
  Thanks for contributing, please follow the instructions in these comments.
  
  This is an Open Source project - please consider contributing a bug fix
  yourself (please read `CONTRIBUTING.md` before starting any work).
-->


### Description

I was following the documentation guide section `Job splits`: https://autosubmit.readthedocs.io/en/master/userguide/defining_workflows/index.html and I've encountered the following error when trying to run the first split experiment with the jobs configuration provided:

```
JOBS:
  INI:
    FILE: INI.sh
    RUNNING: ONCE

  SIM:
    FILE: SIM.sh
    DEPENDENCIES: INI SIM-1
    RUNNING: ONCE

  ASIM:
    FILE: ASIM.sh
    DEPENDENCIES: SIM
    RUNNING: ONCE
    SPLITS: 3

  POST:
    FILE: POST.sh
    RUNNING: ONCE
    DEPENDENCIES:
      ASIM:
        SPLITS_FROM:
          2,3: # [2:3] is also valid
            SPLITS_TO: 1,2*,3* # 1,[2:3]* is also valid, you can also specify the step with [2:3:step]
    SPLITS: 3
```

* Autosubmit version: 4.1.16
* Machine/VM/environment name (if applicable): local
* Command: `autosubmit run a03e`
* Error log:

```
Loading JobList
Load finished
Creating jobs...
Adding dependencies to the graph..
Adding dependencies to the job..
Transitive reduction...
Looking for edgeless jobs...
Autosubmit is running with v4.1.16
Checking the connection to all platforms in use
[local] Correct user privileges for host localhost
[local] Connection successful to host localhost
[WARNING] Known issue: Due to recent changes in Autosubmit's script generation, error line numbers in `script.cmd.err` files may be offset by ~5 lines. Please adjust accordingly when debugging.
Process local_log_recovery started with pid 340551


8 of 8 jobs remaining (13:56)
No jobs to check for platform local
Calculating possible ready jobs for local
Section:INI can submit 20 jobs at this time
Job: a03e_INI submitted with job_id: 340566 and workflow commit: 
a03e_INI submitted
local(log_recovery): Starting...
local(log_recovery): successfully connected.


8 of 8 jobs remaining (13:57)
Checking 1 jobs for platform local
Job a03e_INI is COMPLETED
Calculating possible ready jobs for local
Section:SIM can submit 20 jobs at this time
Job: a03e_SIM submitted with job_id: 340644 and workflow commit: 
a03e_SIM submitted


7 of 8 jobs remaining (13:57)
Checking 1 jobs for platform local
Job a03e_SIM is COMPLETED
Calculating possible ready jobs for local
[ERROR] Arguments provided: Namespace(version=False, logfile='DEBUG', logconsole='WARNING', command='run', expid='a03e', notransitive=False, update_version=False, start_time=None, start_after=None, run_only_members=None, profile=False)
[ERROR] This is the experiment: <a03e> which had an issue with the command: <run> and it is currently using the Autosubmit Version: 4.1.16.
 [CRITICAL] Traceback (most recent call last):
  File "/home/ntorquet/Documents/autosubmit/autosubmit/autosubmit/scripts/autosubmit.py", line 105, in main
    return_value = Autosubmit.run_command(args)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ntorquet/Documents/autosubmit/autosubmit/autosubmit/autosubmit.py", line 838, in run_command
    return Autosubmit.run_experiment(args.expid, args.start_time, args.start_after, args.run_only_members,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ntorquet/Documents/autosubmit/autosubmit/autosubmit/autosubmit.py", line 2251, in run_experiment
    Autosubmit.submit_ready_jobs(as_conf, job_list, platforms_to_test, packages_persistence,
  File "/home/ntorquet/Documents/autosubmit/autosubmit/autosubmit/autosubmit.py", line 2529, in submit_ready_jobs
    packages_to_submit = packager.build_packages()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ntorquet/Documents/autosubmit/autosubmit/autosubmit/job/job_packager.py", line 550, in build_packages
    job.update_parameters(self._as_config, set_attributes=True)
  File "/home/ntorquet/Documents/autosubmit/autosubmit/autosubmit/job/job.py", line 2309, in update_parameters
    parameters = self.update_job_parameters(as_conf, parameters, set_attributes)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ntorquet/Documents/autosubmit/autosubmit/autosubmit/job/job.py", line 2178, in update_job_parameters
    parameters = self.calendar_split(as_conf, parameters, set_attributes)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ntorquet/Documents/autosubmit/autosubmit/autosubmit/job/job.py", line 2036, in calendar_split
    split_start = chunk_start_date(self.date_split, int(self.split), split_length, split_unit, cal)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ntorquet/Documents/autosubmit/autosubmit_env/lib/python3.12/site-packages/bscearth/utils/date.py", line 243, in chunk_start_date
    result = add_time(date, total_months, chunk_unit, cal)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ntorquet/Documents/autosubmit/autosubmit_env/lib/python3.12/site-packages/bscearth/utils/date.py", line 52, in add_time
    return add_days(date, total_size, cal)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ntorquet/Documents/autosubmit/autosubmit_env/lib/python3.12/site-packages/bscearth/utils/date.py", line 106, in add_days
    result = date + relativedelta(days=number_of_days)
             ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
TypeError: unsupported operand type(s) for +: 'NoneType' and 'relativedelta'

 [CRITICAL] Unexpected error: unsupported operand type(s) for +: 'NoneType' and 'relativedelta'.
 Please report it to Autosubmit Developers through Git: https://github.com/BSC-ES/autosubmit/issues
More info at https://autosubmit.readthedocs.io/en/master/troubleshooting/error-codes.html
Retrieving log files a03e_INI.20260331135650.out and .err
a03e_INI_STAT_0 file have been transferred
local(log_recovery) Successfully recovered log for job 'a03e_INI' and retry '0'.
Retrieving log files a03e_SIM.20260331135700.out and .err
a03e_SIM_STAT_0 file have been transferred
local(log_recovery) Successfully recovered log for job 'a03e_SIM' and retry '0'.
local(log_recovery): Exiting.
```
It seems like it's not recognizing the jobs as only-run-once and is assigning them an empty date.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -13,6 +13,7 @@
 - Added priority and retro-compatibility for etc/autosubmitrc over etc/.autosubmitrc #2312
 - Updated Slurm wallclock truncation, allowing users to use e.g., 01:00:00 correctly #2059
 - Fixed exception trace in AutosubmitCritical and AutosubmitError not being displayed to the user #2109
+- Fix `Jobs.running` to case-insensitive #2916
 
 **Enhancements:**
 

--- a/autosubmit/job/job.py
+++ b/autosubmit/job/job.py
@@ -1968,7 +1968,7 @@ class Job(object):
             self.splits = as_conf.jobs_data.get(self.section, {}).get("SPLITS", None)
         self.delete_when_edgeless = as_conf.jobs_data.get(self.section, {}).get("DELETE_WHEN_EDGELESS", True)
         self.dependencies = str(as_conf.jobs_data.get(self.section, {}).get("DEPENDENCIES", ""))
-        self.running = as_conf.jobs_data.get(self.section, {}).get("RUNNING", "once")
+        self.running = str(as_conf.jobs_data.get(self.section, {}).get("RUNNING", "once")).lower()
         self.platform_name = as_conf.jobs_data.get(self.section, {}).get("PLATFORM",
                                                                          as_conf.experiment_data.get("DEFAULT", {}).get(
                                                                              "HPCARCH", None))

--- a/docs/source/userguide/defining_workflows/code/jobs_splits.yml
+++ b/docs/source/userguide/defining_workflows/code/jobs_splits.yml
@@ -1,22 +1,22 @@
 JOBS:
   INI:
     SCRIPT: 'sleep 0'
-    RUNNING: ONCE
+    RUNNING: once
 
   SIM:
     SCRIPT: 'sleep 0'
     DEPENDENCIES: INI SIM-1
-    RUNNING: ONCE
+    RUNNING: once
 
   ASIM:
     SCRIPT: 'sleep 0'
     DEPENDENCIES: SIM
-    RUNNING: ONCE
+    RUNNING: once
     SPLITS: 3
 
   POST:
     SCRIPT: 'sleep 0'
-    RUNNING: ONCE
+    RUNNING: once
     DEPENDENCIES:
       ASIM:
         SPLITS_FROM:

--- a/docs/source/userguide/defining_workflows/index.rst
+++ b/docs/source/userguide/defining_workflows/index.rst
@@ -530,22 +530,22 @@ There is also an special character '*' that can be used to specify that the spli
     JOBS:
       INI:
         FILE: INI.sh
-        RUNNING: ONCE
+        RUNNING: once
 
       SIM:
         FILE: SIM.sh
         DEPENDENCIES: INI SIM-1
-        RUNNING: ONCE
+        RUNNING: once
 
       ASIM:
         FILE: ASIM.sh
         DEPENDENCIES: SIM
-        RUNNING: ONCE
+        RUNNING: once
         SPLITS: 3
 
       POST:
         FILE: POST.sh
-        RUNNING: ONCE
+        RUNNING: once
         DEPENDENCIES:
           ASIM:
             SPLITS_FROM:

--- a/test/unit/test_job.py
+++ b/test/unit/test_job.py
@@ -2228,18 +2228,13 @@ def test_retrieve_logfiles(local, mocker, output):
     job.retrieve_logfiles()
     assert job.log_recovered
 
+
 def test_case_insensitive_running_parameter(autosubmit_config):
     """Test that the ``running`` parameter is case-insensitive."""
-    as_conf = autosubmit_config(_EXPID, {
-        'JOBS': {
-            'A': {
-                'FILE': 'a',
-                'PLATFORM': 'local',
-                'RUNNING': 'ONCE'
-            }
-        }
-    })
-    job = Job(_EXPID, '1', status=Status.WAITING, priority=0, loaded_data=None)
-    job.section = 'A'
+    as_conf = autosubmit_config(
+        _EXPID, {"JOBS": {"A": {"FILE": "a", "PLATFORM": "local", "RUNNING": "ONCE"}}}
+    )
+    job = Job(_EXPID, "1", status=Status.WAITING, priority=0, loaded_data=None)
+    job.section = "A"
     job.update_dict_parameters(as_conf)
-    assert job.running == 'once'
+    assert job.running == "once"

--- a/test/unit/test_job.py
+++ b/test/unit/test_job.py
@@ -2227,3 +2227,19 @@ def test_retrieve_logfiles(local, mocker, output):
     job.platform.check_file_exists = mocker.MagicMock(return_value=True)
     job.retrieve_logfiles()
     assert job.log_recovered
+
+def test_case_insensitive_running_parameter(autosubmit_config):
+    """Test that the ``running`` parameter is case-insensitive."""
+    as_conf = autosubmit_config(_EXPID, {
+        'JOBS': {
+            'A': {
+                'FILE': 'a',
+                'PLATFORM': 'local',
+                'RUNNING': 'ONCE'
+            }
+        }
+    })
+    job = Job(_EXPID, '1', status=Status.WAITING, priority=0, loaded_data=None)
+    job.section = 'A'
+    job.update_dict_parameters(as_conf)
+    assert job.running == 'once'

